### PR TITLE
Main functionality of join is done.

### PR DESCRIPTION
**Pull request by Prastaruszek**
Main functionality of join is done #271. What is more -t, -j, -a options are ready. The code can be easly extended with the implementation of rest of the flags.

### CODE REVIEW

**Comment by ebfe** [comment]
Hi,

this seems to behave differently form GNU join. For example:

```
$ cat a
1 foo1 1
2 foo2 1
$ cat b
1 bar1 1
2 bar2 2
# GNU join
$ join -1 3 a b
1 1 foo1 bar1 1
1 2 foo2 bar1 1
# uutils
$ build/join -1 3 a b                                                                                                                                      1 1 foo1 bar1 1
$ 
```

Maybe you could add a few tests to check against known good output from GNU join?

**Comment by Phlosioneer** [comment]
Any update on this?

File: src/join/join.rs [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+        let second_splited: Vec<&str>;
+        let space = " ";
+        let other = &opts.separator.to_string();
+        let sep: &str;
+        if opts.white_spc_sep {
+            first_splited = first.split_whitespace().collect();
+            second_splited = second.split_whitespace().collect();
+            sep = space;
+        }
+        else{
```
Comment (odeits) [inline src/join/join.rs]: shouldn't there be a space? `else {`

File: src/join/join.rs [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+            first_splited = first.split(opts.separator).collect();
+            second_splited = second.split(opts.separator).collect();
+            sep = other;
+        }
+        if first_splited.len() > opts.first_field
+        && second_splited.len() > opts.second_field
+        && first_splited.get(opts.first_field).unwrap()
+           == second_splited.get(opts.second_field).unwrap() {
+            let mut i=0;
```
Comment (pickfire) [inline src/join/join.rs]: Shouldn't it be `let mut i = 0;`?

File: src/join/join.rs [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+               first_splited.get(opts.first_field)
+                    .unwrap().to_string();
+            for x in first_splited {
+                if i!=opts.first_field {
+                    out_string.push_str(sep);
+                    out_string.push_str(x);
+                }
+                i = i+1;
+            }
+            i=0;
```
Comment (pickfire) [inline src/join/join.rs]: Isn't it `i = 0;` here?

File: src/join/join.rs (hunk 1 above)
Comment (pickfire) [reply src/join/join.rs]: Nope, I think it should be `} else {`.

File: src/join/join.rs [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+            for x in second_splited {
+                if i!= opts.second_field {
+                    out_string.push_str(sep);
+                    out_string.push_str(x);
+                }
+                i = i+1
```
Comment (pickfire) [inline src/join/join.rs]: Did you mean `i += 1;` here?

File: src/join/join.rs [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+            for x in first_splited {
+                if i!=opts.first_field {
+                    out_string.push_str(sep);
+                    out_string.push_str(x);
+                }
+                i = i+1;
+            }
+            i=0;
+            for x in second_splited {
+                if i!= opts.second_field {
```
Comment (pickfire) [inline src/join/join.rs]: Shouldn't it be `if i != opts.second_field {`?

**Comment by timofonic** [comment]
@Prastaruszek @ebfe @Phlosioneer @odeits @Arcterus Any update on this?

**Pull request closed without merging**